Cleanup

azz opened the pull request:
* Cleanup unused AST nodes
* Add tests for JSXNamespacedName
* **Turn on importMeta babylon plugin**
* Add tests for DeclareTypeAlias

This actually caught a minor issue, we didn't have the `importMeta` babylon plugin turned on. [repro](https://prettier.io/playground/#%7B%22content%22%3A%22const%20x%20%3D%20import.meta%3B%5Cnconst%20url%20%3D%20import.meta.url%3B%5Cnimport.meta%3B%5Cnimport.meta.url%3B%5Cnimport.meta.couldBeMutable%20%3D%20true%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Atrue%7D%7D)

There's still a lot of `TS*Keyword` nodes that _are_ used, though, so I'm not sure if its 100% safe to remove all the TS nodes that I did.

vjeux commented:
Could you run this against the public typescript repos and see if it catches anything?

vjeux commented:
Discussion that led to this PR: https://github.com/prettier/prettier/pull/2497#issuecomment-315603289

azz commented:
Yeah I already ran it against the TypeScript tests suite. Some of those `TS*Keywords` are needed so I added them back.

I also found and fixed an issue with destructuring.

We lost the `?` in both these cases:

```ts
function f({ x }?) {}
function g([y]?) {}
```

azz commented:
![image](https://user-images.githubusercontent.com/1297597/28248134-a85bb6c4-6a82-11e7-9bd0-78392471a7fd.png)

Neat.
